example/lib/main.dart

```diff
@@ -480,11 +481,11 @@ class _MyAppState extends State<MyApp> {
       );
 
   Widget getPageNavigator<T>({
-    String name,
-    bool enabled,
-    ably.PaginatedResult<T> page,
-    Future<ably.PaginatedResult<T>> Function() query,
-    Function(ably.PaginatedResult<T> result) onUpdate,
+    required bool enabled,
+    String? name,
+    ably.PaginatedResult<T>? page,
+    Future<ably.PaginatedResult<T>> Function()? query,
+    Function(ably.PaginatedResult<T> result)? onUpdate,
```

Comment:
These 2 parameters (query, onUpdate) are always provided in the usages of `getPageNavigator`. They are also required in the logic of this method. Why not make them required instead of optional? Then we can remove the `!` from onUpdate and query.

Reply:
ff2a1ce4cc50d6cb75664e8256a083a28ae3445c